apps/zui/scripts/merge-mac-release-files.mjs

```diff
@@ -0,0 +1,227 @@
+// Based on https://github.com/electron-userland/electron-builder/issues/5592#issuecomment-2004803764
+import { Octokit } from "octokit";
+import pkg from '../package.json' assert {type: 'json'};
+import { TextDecoder } from 'node:util';
+import fs from 'node:fs';
+import { Readable } from 'node:stream';
+import { Buffer } from 'node:buffer';
+import yaml from 'js-yaml';
+
+const token = process.env.GH_TOKEN
+
+const client = new Octokit({
+  auth: token
+});
+
+// These are derived from settings in package.json so the script will work on
+// both regular Zui and Zui Insiders.
+const OWNER = pkg.repository.split('/')[3];
+const REPO = pkg.repository.split('/')[4];
+const PRODUCT_NAME= pkg.productName.replaceAll(' ', '-');
```

Comment:
This `replaceAll` was needed because the build filenames that start out with space characters in them (e.g., `Zui - Insiders-1.7.1-19.dmg`) have those spaces turned into hyphens by the time they're uploaded to GItHub. I briefly explored using this as an opportunity drop those space characters entirely, but post-update that ends up putting the new app state & saved data in a fresh, new directory `~/Library/Application Support/Zui-Insiders` hence losing the prior state/data from the traditional `~/Library/Application Support/Zui - Insiders`, hence that would require a new migration logic for users to have a seamless update. Yuck! Hence the `replaceAll` to just keep going with what we've got.